lib/core/src/persist/cache.rs

```diff
@@ -64,17 +65,81 @@ impl Persister {
         res
     }
 
-    pub fn set_wallet_info(&self, info: &GetInfoResponse) -> Result<()> {
+    pub fn set_wallet_info(&self, info: &WalletInfo) -> Result<()> {
         let serialized_info = serde_json::to_string(info)?;
         self.update_cached_item(KEY_WALLET_INFO, serialized_info)
     }
 
-    pub fn get_wallet_info(&self) -> Result<Option<GetInfoResponse>> {
-        let info_str = self.get_cached_item(KEY_WALLET_INFO)?;
-        Ok(match info_str {
-            Some(str) => serde_json::from_str(str.as_str())?,
-            None => None,
-        })
+    pub fn set_blockchain_info(
+        &self,
+        liquid_tip: Option<u32>,
+        bitcoin_tip: Option<u32>,
+    ) -> Result<()> {
+        let con = self.get_connection()?;
+
+        let mut blockchain_info = serde_json::Map::new();
+        if let Some(liquid_tip) = liquid_tip {
+            blockchain_info.insert("liquid_tip".to_string(), liquid_tip.into());
+        }
+        if let Some(bitcoin_tip) = bitcoin_tip {
+            blockchain_info.insert("bitcoin_tip".to_string(), bitcoin_tip.into());
+        }
+
+        con.execute(
```

Comment:
Why not use update_cached_item as with other settings?

Reply:
We were doing partial updates before, but I thought about it and decided it's better so update the whole object info instead (only when the fetch from both tips succeeds)